### PR TITLE
enable instance network configuration WRT gnt-network

### PR DESCRIPTION
**Pull request by saschalucas**
With the introduction of gnt-network all information regarding the
configration of an instance NIC is present. This patch makes use of it, so
that the instance comes up online (ready to connect via SSH). Instances must
be added with "gnt-instance add --net 0:network=<name>,ip=<pool_or_ip ..."

### CODE REVIEW

**Comment by saschalucas** [comment]
I signed it!

File: examples/hooks/interfaces [hunk 1]
```diff
@@ -38,6 +43,25 @@ fi
 
 if [ "$NIC_COUNT" -gt 0 ]; then
 
+# check weather to use dhcp or static method
```
Comment (iustin) [inline examples/hooks/interfaces]: Typo weather → whether.

**Review by iustin — COMMENTED** [review]
Yay, first pull request on instance-debootstrap!

I have two comments.

1) There's a lot of ${..} in the file, for simple variables (even when there can be no confusion over the intended meaning. As you can see, the file already uses non-braced expressions ($NIC_COUNT, $TARGET), so would prefer to keep same style.

2) The patch seems more flexible than needed, by defining per-type config blocks, etc. I see a much simpler way:

```
cat > $TARGET/etc/network/interfaces <<EOF
 auto lo
 iface lo inet loopback
 
 auto eth0
EOF

if [ -n "$NIC_0_IP" ]; then
 ...
 cat >> $TARGET/etc/network/interfaces <<EOF
    iface eth0 inet dhcp
    address ...
    netmask ...
    gateway ...
EOF
else
  cat >> $TARGET/etc/network/interfaces <<EOF
    iface eth0 dhcp
EOF
```

The code flow would be somewhat simpler to follow. I'm fine either way, so what do you think?

**Comment by saschalucas** [comment]
Hi  iustin,

I've tried to match the variables style, where possible and also change the flow as suggested.

Thanks, Sascha.

File: examples/hooks/interfaces [hunk 2]
```diff
@@ -46,8 +49,28 @@ auto lo
...
+      exit 1
+    fi
+    if [ -z "$NIC_0_NETWORK_GATEWAY" ]; then
+      echo "Missing NETWORK GATEWAY from gnt-network"
+      exit 1
+    fi
+    cat >> $TARGET/etc/network/interfaces <<EOF
+iface eth0 inet static
+  address $NIC_0_IP
+  netmask ${NIC_0_NETWORK_SUBNET##*/}
```
Comment (iustin) [inline examples/hooks/interfaces:66]: Question: I don't know what value NIC_0_NETWORK_SUBNET will have. Is is something like 192.168.0.0/16? If so, does it mean that here will we have "netmask 16"?

If so, I didn't know this works. The `ifup` command behaves differently between 255.255.255.0 and 24, for some strange reason: in the first case, passes explicit broadcast address to `ip`, in the second, it passes `+`, which has special significance (auto-compute). I wonder why it is different… anyway, it works.

**Review by iustin — APPROVED** [review]

**Pull request merged**